typescript/phoenix-sdk/src/market.ts

```diff
@@ -197,76 +318,319 @@ export class Market {
+  /**
+   * Returns a Phoenix swap order packet
+   *
+   * @param market The market to submit the order to
+   * @param side The side of the order
+   * @param inAmount The amount of the input token
+   * @param slippage The slippage tolerance in bps (optional, default 0.5%)
+   * @param selfTradeBehavior The self trade behavior (optional, default Abort)
+   * @param matchLimit The match limit (optional)
+   * @param clientOrderId The client order ID (optional)
+   * @param useOnlyDepositedFunds Whether to use only deposited funds (optional)
+   */
+  getSwapOrderPacket({
+    side,
+    inAmount,
+    slippage = DEFAULT_SLIPPAGE_PERCENT,
+    selfTradeBehavior = SelfTradeBehavior.Abort,
+    matchLimit = DEFAULT_MATCH_LIMIT,
+    clientOrderId = 0,
+    useOnlyDepositedFunds = false,
+  }: {
+    side: Side;
+    inAmount: number;
+    slippage?: number;
+    selfTradeBehavior?: SelfTradeBehavior;
+    matchLimit?: number;
+    clientOrderId?: number;
+    useOnlyDepositedFunds?: boolean;
+  }): Partial<OrderPacket> {
+    const expectedOutAmount = this.getExpectedOutAmount({
+      side,
+      inAmount,
+    });
+    const baseMul = 10 ** this.baseToken.data.decimals;
+    const quoteMul = 10 ** this.quoteToken.data.decimals;
+    const slippageDenom = 1 - slippage;
+    let numBaseLots = 0;
+    let minBaseLotsToFill = 0;
+    let numQuoteLots = 0;
+    let minQuoteLotsToFill = 0;
+
+    if (side === Side.Ask) {
+      numBaseLots =
+        (inAmount * baseMul) /
+        parseFloat(this.data.header.baseLotSize.toString());
+      minQuoteLotsToFill = Math.ceil(
+        ((expectedOutAmount * quoteMul) /
+          parseFloat(this.data.header.quoteLotSize.toString())) *
+          slippageDenom
+      );
+    } else {
+      numQuoteLots =
+        (inAmount * quoteMul) /
+        parseFloat(this.data.header.quoteLotSize.toString());
+      minBaseLotsToFill = Math.ceil(
+        ((expectedOutAmount * baseMul) /
+          parseFloat(this.data.header.baseLotSize.toString())) *
+          slippageDenom
+      );
+    }
+
+    const order: Partial<OrderPacket> = {
+      side,
+      priceInTicks: null,
+      numBaseLots,
+      minBaseLotsToFill,
+      numQuoteLots,
+      minQuoteLotsToFill,
+      selfTradeBehavior,
+      matchLimit,
+      clientOrderId,
+      useOnlyDepositedFunds,
+    };
+
+    return order;
+  }
+
+  /**
+   * Returns the expected amount out for a given swap order
+   *
+   * @param market The market to calculate the amount out for
+   * @param side The side of the order (Bid or Ask)
+   * @param inAmount The amount of the input token
+   */
+  getExpectedOutAmount({
+    side,
+    inAmount,
+  }: {
+    side: Side;
+    inAmount: number;
+  }): number {
+    const numBids = toNum(this.data.header.marketSizeParams.bidsSize);
+    const numAsks = toNum(this.data.header.marketSizeParams.asksSize);
+    const ladder = this.getUiLadder(Math.max(numBids, numAsks));
```

Comment:
This should use the regular `getLadder` instead of `getUiLadder` so we aren't doing the units math multiple times